### PR TITLE
fix(dropdown): fix dropdown inside click

### LINKED ISSUE #1933: fix(dropdown) autoClose on outside-click doesn't work with unallowed autoClose on inside-click
Hi,

I just updated to angular4, so I had to update ngx-bootstrap as well.
I use the dropdowns quite a lot and was able to do all the upgrades needed, to get them working again, BUT the autoClose="onOutsideClick" feature seems to be gone. That was sooo important, why would you get rid of it?

Is there any way to get this functionality back?

Thanks a bunch!
Tom

### PR DESCRIPTION
**Pull request by IlyaSurmay**
Fixes #1933 

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated API documentation.
 - [x] added/updated demos.

### CODE REVIEW

**Comment by daddykotex** [comment]
Hi, 

IMO this is a very nice feature. Do you need any help for this? I'm willing to help to get this merged!

Thanks to @IlyaSurmay

**Comment by valorkin** [comment]
I am slowly getting there
~50 PRs already merged, 50 to go :D

**Comment by daddykotex** [comment]
Oh ok, no issue, If I can do anything, let me know

File: src/dropdown/bs-dropdown-container.component.ts [hunk 1]
```diff
@@ -35,11 +35,11 @@ export class BsDropdownContainerComponent implements OnDestroy {
     private _state: BsDropdownState,
     private cd: ChangeDetectorRef,
     private _renderer: Renderer2,
-    _element: ElementRef
+    public _element: ElementRef
```
Comment (valorkin) [inline src/dropdown/bs-dropdown-container.component.ts]: `public`?

File: src/dropdown/bs-dropdown-toggle.directive.ts [hunk 2]
```diff
@@ -50,12 +51,17 @@ export class BsDropdownToggleDirective implements OnDestroy {
 
   @HostListener('document:click', ['$event'])
   onDocumentClick(event: any): void {
-    if (
-      this._state.autoClose &&
-      event.button !== 2 &&
-      !this._element.nativeElement.contains(event.target)
-    ) {
-      this._state.toggleClick.emit(false);
+    if (this._state.autoClose && event.button !== 2 && !this._element.nativeElement.contains(event.target)) {
```
Comment (valorkin) [inline src/dropdown/bs-dropdown-toggle.directive.ts]: both if's inside do the same thing?

File: src/dropdown/bs-dropdown.directive.ts [hunk 3]
```diff
@@ -113,21 +118,21 @@ export class BsDropdownDirective implements OnInit, OnDestroy {
...
+  private _isInlineOpen = false;
 
+  private _inlinedMenu: EmbeddedViewRef<BsDropdownMenuDirective>;
   private _isDisabled: boolean;
-  private _dropdown: ComponentLoader<BsDropdownContainerComponent>;
   private _subscriptions: Subscription[] = [];
   private _isInited = false;
 
-  constructor(private _elementRef: ElementRef,
+  constructor(public _elementRef: ElementRef,
```
Comment (valorkin) [inline src/dropdown/bs-dropdown.directive.ts]: why public?

**Review by valorkin — CHANGES_REQUESTED** [review]
don't make _elRefs public
add `_contains` method and mark it as `@internal` in jsdoc

**Comment by Domainv** [comment]
Close to prefer #4609

**Pull request closed without merging**